### ISSUE TITLE
BeanCreationException while launching the jenkins-codequality-3.1.0-SNAPSHOT.jar

### ISSUE THREAD

**Issue by sunabala** [Hygieia/hygieia-codequality-jenkins-collector#5]
[application.properties.txt](https://github.com/Hygieia/hygieia-codequality-jenkins-collector/files/3552152/application.properties.txt)
Hi,

I built hygieia-codequality-jenkins-collector and deployed using the below command:
java -jar jenkins-codequality-3.1.0-SNAPSHOT.jar --spring.config.name=jenkins-codequality --spring.config.location=..\application.properties
I am getting the BeanCreationException.  Attached is my application.properties file and the error message.  Please help.
![image](https://user-images.githubusercontent.com/32274756/63879045-b0750780-c97f-11e9-88bb-6609de2a79f1.png)
![image](https://user-images.githubusercontent.com/32274756/63879063-b9fe6f80-c97f-11e9-9b42-8af547ac61aa.png)
![image](https://user-images.githubusercontent.com/32274756/63879069-bf5bba00-c97f-11e9-800e-b891b0e3c851.png)

Thanks,
Sunandha Ramasamy

**Comment by rvema** [comment]
@sunabala looks like its unable to locate the properties file. Can you give full path to properties file while launching the collector

**Comment by sunabala** [comment]
@rvema, I had provided the complete path to properties file while launching the collector... Still the issue exists.

**Comment by sunabala** [comment]
On further debugging, found that this.settings.getCron() call in public String getCron() method of JenkinsCodeQualityCollectorTask.java is null.  What needs to be done for this to work?

**Comment by rvema** [comment]
the cron property is not properly formed or the property file is unable to be read. Can you check if other properties mentioned in the application.properties are read?

**Comment by sunabala** [comment]
Yes it is reading the application.properties file.  It is able to establish the connection with MONGO DB based on the DB credentials provided in the properties file.
This is the content of the properties file:
# Database Name
dbname=dashboarddb
# Database HostName - default is localhost
dbhost=localhost
# Database Port - default is 27017
dbport=27017
# MongoDB replicaset
dbreplicaset=false 
dbhostport=localhost:27017
# Database Username - default is blank
dbusername=dashboarduser
# Database Password - default is blank
dbpassword=dbpassword
# Collector schedule (required)
jenkins-codequality.cron=0 0/1 * * * *
# Collector servers (required) - can be multiple
jenkins.servers[0]=https://jenkins.company.com
# Collector types (note not required, but the regex should be match only the type specified)
 jenkins-codequality.artifactRegex.junit=TEST-.*\\.xml
 jenkins-codequality.artifactRegex.findbugs=findbugsXml.xml
 jenkins-codequality.artifactRegex.pmd=pmd.xml
 jenkins-codequality.artifactRegex.checkstyle=checkstyle-result.xml
 jenkins-codequality.artifactRegex.jacoco=jacoco.xml
# Collector job depth (required) should be set to at least 1, and more if you use folder jobs etc
jenkins-codequality.jobDepth=4
